### ISSUE TITLE
Redundant information about Submodels in AAS Registry, Submodel Registry and AAS

### ISSUE THREAD

**Issue by FrankSchnicke** [admin-shell-io/aas-specs-api#114]
# What?
An AAS contains references to its submodels. At the same time, an AAS descriptor contains submodel descriptors, thus implicitly providing information about the submodels associated with the AAS. In addition, a separate submodel registry may contain a descriptor for the same submodel.
Consequently, (a) the "AAS -> Submodel" relationship information is redundant between the AAS and the AAS Registry, and (b) the "SubmodelId -> SubmodelDescriptor" information is redundant between the AAS Registry and the Submodel Registry.

This leads to the following problems:
- Inconsistent state of the "AAS -> Submodel" relationship if the submodel references of an AAS are changed but the AAS registry is not updated, and vice versa. This can happen not only due to oversight, but also due to simple timing issues, as it can't be guaranteed that the AAS & registry updates happen at the same time.
- Inconsistent state of "SubmodelId -> SubmodelDescriptor" information due to the AAS registry being updated (e.g. with a new submodel endpoint due to a deployment change) and the submodel registry not being updated and vice versa. Again, this can happen not only due to oversight, but also due to simple timing issues, as it can't be guaranteed that the AAS registry & submodel registry updates happen at the same time.

![image](https://github.com/admin-shell-io/aas-specs-api/assets/77283144/79f2b76e-c211-4b4b-acf8-f1d70019055b)

# How?
Clarify where the ground truth is for "AAS -> Submodel" relationship information as well as "SubmodelId -> SubmodelDescriptor" information.
From my point of view, it would be best to define a clear retrieval sequence (see figure below) and mark the use of SubmodelDescriptors in AASDescriptors as deprecated.

![image](https://github.com/admin-shell-io/aas-specs-api/assets/77283144/f9074da5-9912-4a27-9751-d6c9a25bb65b)

**Comment by BirgitBoss** [comment]
> > Please have a look at the note in the Part 2 specification on the class AssetAdministrationShellDescriptor
> 
> Thanks for pointing out this remark, I somehow missed it. However, in my opinion, there's still the issue on how to ensure that the SubmodelDescriptors contained in the AASDescriptor are synchronized with the SubmodelDescriptors contained in the Submodel Registry for Submodels with the same Id. Or is it expected that the SubmodelDescriptors are only provided by the AAS Registry if it somehow ensures this internally?

Exactly, it is not expected that you provide both, an AAS Registry with submodel descriptors and a Submodel registry as stand-alone implementation providing submodel desriptors as well.
